feat(theme): Force `color` to black by default, add theming option

tecc opened the pull request:
better-theming: The CSS selector now also applies `color: black` to go with `background-color: white`.
option-force-light-theme: A new option has been added that forces the colours to always be white for the foreground and black for the background, otherwise inheriting the colours. It is on by default to be as close to original behaviour as possible.

tecc committed:
feat(theme): Force `color` to black by default, add theming option

better-theming: The CSS selector now also applies `color: black` to go with `background-color: white`.
option-force-light-theme: A new option has been added that forces the colours to always be white for the foreground and black for the background, otherwise inheriting the colours. It is on by default to be as close to original behaviour as possible.

tessus commented:
Thanks for your contribution. With PRs like these "before" and "after" screenshots help a lot to decide whether it should be merged. This is just my personal opinion, since this is Laurent's plugin, I won't be able to make and decisions anyway.

tecc commented:
Current behaviour:
![bild](https://github.com/joplin/plugin-abc-sheet-music/assets/38840234/0fa6294e-db74-4458-a013-e0793761504a)


New behaviour, `forceLightTheme = false`:
![Inherited theme colours](https://github.com/joplin/plugin-abc-sheet-music/assets/38840234/a70c574c-2ee2-4e7f-ac90-3eaff8e24a32)

New behaviour, `forceLightTheme = true`:
![Forced light theme](https://github.com/joplin/plugin-abc-sheet-music/assets/38840234/693c377a-2c0d-43d9-9404-c830554d5339)

tecc committed:
fix(theme): Reimplementation of theming

theming: The container element now specifies the colours in its `style` attribute. Consequently, the CSS asset specifies to inherit parent colours. It's essentially a shorter way of expressing the same thing.

tessus commented:
I like this change, although the default should probably be the previous behavior, otherwise you break people's current setup. I use the dark theme as well, but in certain situations I like the light theming look better. Having played an instrument in the past, music sheets with anything other than a light background are rather alien to me.
Let's see what Laurent thinks.

tecc commented:
The "Force light theme" option, which forces the black foreground/white background combination, is by default on.

tessus commented:
~~Yes, but it wasn't before this PR, was it? So people who install the new version (after this PR is merged) will see the forced white background. This "breaks" their setup.
When you add a feature it should not change the default behavior (except in rare circumstances)~~

I misunderstood your sentence. All is good.

laurent22 commented:
Sorry I may have missed this PR. ABC is now part of the main app - if this feature is still not supported and you'd like to add it please move the PR there. The code for ABC is there: https://github.com/laurent22/joplin/tree/dev/packages/renderer/MdToHtml/rules
